### PR TITLE
chore: rename assert to test for markdown syntax

### LINKED ISSUE #1399: chore: update snapshot tests syntax
use `test` instead of `assert` in the md files.

**Current**
<pre>
```yml @assert
- method: POST
  url: http://localhost:8080/graphql
  body:
    query: query { username }
```
</pre>

**Proposed**
<pre>
```yml @test
- method: POST
  url: http://localhost:8080/graphql
  body:
    query: query { username }
```
</pre>

### PR DESCRIPTION
**Pull request by timburrows**
**Summary:**  
Renames `assert` blocks to `test` blocks in snapshot test markdown. 

**Issue Reference(s):**  
Fixes #1399 

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [x] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [] I have added relevant unit & integration tests.
- [x] I have updated the [documentation] accordingly.
- [x] I have performed a self-review of my code.
- [x] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

**Comment by tusharmath** [comment]
Need to fix the build. Some tests are failing, I think that's because there are some residual `assert` statements.

**Comment by tusharmath** [comment]
@timburrows Moving it to draft. Please mark it as ready to review once the build passes.

**Comment by timburrows** [comment]
Thanks. Changes from upstream introduced more `assert` blocks. Workflows report that it requires approval before builds will trigger

**Comment by timburrows** [comment]
This change would invalidate documentation found here https://github.com/tailcallhq/tailcallhq.github.io/blob/develop/docs/contributors/tests.mdx. Should another issue be raised for this?

**Comment by tusharmath** [comment]
@timburrows Documentation should be part of this fix.

**Comment by timburrows** [comment]
@tusharmath see this PR for the aforementioned changes https://github.com/tailcallhq/tailcallhq.github.io/pull/208

File: tests/README.md [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+Each test's `.md` file may have a file name suffix of `-error` if the bare file name is already taken by a previously converted test.
+
+1. The name is generated from the file name, and put into a [header](#header).
+1. An [`sdl error` instruction](#instruction) is appended.
+1. The server SDL is put into a [`server` block](#server).
+1. The expected errors are put into an `errors` snapshot.
+
+## Maintenance
+
+1. To clean unused snapshots, run `cargo insta test --delete-unreferenced-snapshots`.
```
Comment (tusharmath) [inline tests/README.md:293]: We don't need this file. Changes should be incorporated https://github.com/tailcallhq/tailcallhq.github.io/pull/208

**Review by tusharmath — APPROVED** [review]

**Review by tusharmath — APPROVED** [review]

**Review by amitksingh1490 — APPROVED** [review]

**Comment by amitksingh1490** [comment]
@timburrows  Thanks for your great contribution to Tailcall! 🌟 Your expertise is invaluable. 🙌 We'd love to see you engage more with our community and continue to drive innovation. 👍

**Pull request merged**